pkg/tsdb/grafana-pyroscope-datasource/pyroscopeClient.go

```diff
@@ -108,6 +108,7 @@ func (c *PyroscopeClient) GetSeries(ctx context.Context, profileTypeID string, l
 		End:           end,
 		Step:          step,
 		GroupBy:       groupBy,
+		Limit:         limit,
```

Comment:
Something is not working...

**Browser**

E.g. `limit = 2`:
![image](https://github.com/user-attachments/assets/3d18f49f-75b8-486b-8273-220ca86cf890)

The response still contains the 4 series:
<img width="791" alt="image" src="https://github.com/user-attachments/assets/a22c8091-5ad4-430f-bcde-ca5320dee48d">

**Data source logs**
```json
{"@level":"debug","@message":"Sending SelectSeriesRequest","@timestamp":"2024-10-10T11:54:57.077255+02:00","function":"/grafana/grafana/pkg/tsdb/grafana-pyroscope-datasource/service.go:50[grafana-pyroscope-datasource.logEntrypoint]","logger":"tsdb.pyroscope","queryModel":{"WithStreaming":false,"datasource":{"type":"grafana-pyroscope-datasource","uid":"de0fruujmm41sb"},"groupBy":["span_name"],"labelSelector":"{service_name=\"pyroscope-rideshare-go\"}","limit":2,"maxNodes":0,"profileTypeId":"process_cpu:cpu:nanoseconds:cpu:nanoseconds","queryType":"metrics","refId":"A"}}
```

Reply:
The explanation is that the test above was made against production and the API changes have not yet reached it. Against a dev instance it works.